WagerMate/Service-Implementation/betting/BetSettlingService.cs

```diff
@@ -0,0 +1,79 @@
+using WagerMate.Data;
+using WagerMate.Service_Implementation.user;
+using WagerMate.Services.betting;
+
+namespace WagerMate.Service_Implementation.betting;
+
+public class BetSettlingService : IBetSettlingService
+{
+    private IBetService _betService;
+    private IUserBetService _userBetService;
+    private IWinnerService _winnerService;
+
+    public BetSettlingService(IBetService betService, IUserBetService userBetService, IWinnerService winnerService)
+    {
+        _betService = betService;
+        _userBetService = userBetService;
+        _winnerService = winnerService;
+    }
+    
+    public bool CloseBet(UserBet userBet)
+    {
+        try
+        {
+            Bet activeBet = _betService.GetBetById(userBet.Bet_Id);
+            if (activeBet.State == State.Closed) return false;
+            activeBet.State = State.Closed;
+            _betService.UpdateBet(activeBet);
+            return true;
+        }
+        catch (Exception)
+        {
+            Console.WriteLine("BetSettlingService: closeBet failed");
+        }
+        return false;
+    }
+
+    public bool SettleBet(Bet bet, int[] caseIds)
```

Comment:
Warum werden als Parameter die caseids benötigt?
```suggestion
    public bool SettleBet(Bet bet)
```

Reply:
Ohne zu wissen, welcher Case ein Gewinner-Case ist, weiß man nicht, welche Winner generiert werden soll. Den Parameternamen werde ich anpassen zu "int[] winningCaseIds"